src/components/token/TokenBalances.js

```diff
@@ -77,20 +90,17 @@ class TokenBalances extends React.Component {
       return;
     }
 
-    // 'Click' on search to make the first query
-    const queryParams = this.pagination.obtainQueryParams();
-
-    await helpers.setStateAsync(this, {
-      sortBy: get(queryParams, 'sortBy', this.state.sortBy),
-      order: get(queryParams, 'order', this.state.order),
-      loading: true,
-    });
-
-    await this.performSearch();
+    if (this.state.tokenId === hathorLibConstants.HATHOR_TOKEN_CONFIG.uid) {
+      // If we had a custom token as queryParam
+      // then we will perform the search after the token
+      // is found in the elastic search
+      // otherwise we just perform the search for HTR to show the default screen
+      await this.performSearch();
```

Comment:
I did not get why the performSearch is called only when HTR is the token. Supposing a scenario where a custom token is given as param, where is it going to be searched?

Reply:
Every time you search for a new token in the autocomplete component, a new search is triggered. What I did was to manually select a token in the autocomplete, then the search will be triggered as soon as we get the token element from the elastic search and add to the autocomplete.

I did like this so we don't need to change any structure we have nowadays, the token in the query param just simulate a search and click.